chain/types/blockheader.go

```diff
@@ -171,7 +174,7 @@ func PowerCmp(eproof ElectionProof, mpow, totpow BigInt) bool {
 	// 2^256
 	rden := BigInt{big.NewInt(0).Exp(big.NewInt(2), big.NewInt(256), nil)}
```

Comment:
IDK when this code came in but I've talked about it with Henri.
We've arrived at:
`h(vrfout) * totalPower < minerPower * 2^256`

This way we avoid division which can introduce a skew.
The fully correct version of this would be `< minerPower *  (2^256-1)` as it is max value of `h(x)`.

This can be implemented as:
```
lhs := BigFromBytes(h[:]).Int
lhs = lhs.Mul(lhs, totpow.Int)
rhs := new(big.Int).Lsh(mpow.Int, 256)
rhs = rhs.Sub(rhs, mpow.Int)
// return compare
```

Reply:
Want to push a fix?